From e3cf8e5f63362a09667b8c9c9df7ac48a9d5231d Mon Sep 17 00:00:00 2001
From: Geetha <gakula@cavium.com>
Date: Mon, 24 Apr 2017 17:29:36 +0530
Subject: [PATCH 5/9] iommu/arm-smmu-v3: Increase SMMU CMD queue poll timeout

Patch-mainline: Submitted, https://www.spinics.net/lists/arm-kernel/msg577745.html
References: bsc#1035479

When large memory is being unmapped, huge no of tlb invalidation cmds are
submitted followed by a SYNC command. This sometimes hits CMD queue full and
poll on queue drain is being timedout throwing error message 'CMD_SYNC timeout'.

Although there is no functional issue, error message confuses user. Hence increased
poll timeout to 500us

Signed-off-by: Geetha <gakula@cavium.com>
Message-Id: <1493035176-3633-1-git-send-email-gakula@caviumnetworks.com>
Signed-off-by: Robert Richter <rrichter@cavium.com>
Signed-off-by: Mian Yousaf Kaukab <yousaf.kaukab@suse.com>
---
 drivers/iommu/arm-smmu-v3.c | 2 +-
 1 file changed, 1 insertion(+), 1 deletion(-)

diff --git a/drivers/iommu/arm-smmu-v3.c b/drivers/iommu/arm-smmu-v3.c
index 558c895d..af63bdfe 100644
--- a/drivers/iommu/arm-smmu-v3.c
+++ b/drivers/iommu/arm-smmu-v3.c
@@ -410,7 +410,7 @@
 #define PRIQ_1_ADDR_MASK		0xfffffffffffffUL
 
 /* High-level queue structures */
-#define ARM_SMMU_POLL_TIMEOUT_US	100
+#define ARM_SMMU_POLL_TIMEOUT_US	500
 
 #define MSI_IOVA_BASE			0x8000000
 #define MSI_IOVA_LENGTH			0x100000
-- 
2.11.0

